### ISSUE TITLE
String representation Response and Take

### ISSUE THREAD

**Issue by dmzaytsev** [yegor256/takes#101]
I expect implementations of Response and Take interfaces will be able convert itself to a loggable string but  they don't have this feature

**Comment by yegor256** [comment]
Something like this you have in mind?

```
interface ConsoleFriendly {
  String toConsole();
}
interface Take extends ConsoleFriendly {
}
```

right?

**Comment by dmzaytsev** [comment]
@yegor256 Yes, something like this will be suitable

**Comment by davvd** [comment]
@yegor256 this task is yours, go ahead

**Comment by davvd** [comment]
@dmzaytsev thank you for reporting this, I added 15 mins to your acc, transaction `54622993`

**Comment by davvd** [comment]
the code made here contains a puzzle `101-0be068b8`, which will be resolved soon

**Comment by davvd** [comment]
@dmzaytsev since there is no milestone yet I set it to "1.0"

**Comment by yegor256** [comment]
@davvd assign someone else pls

**Comment by yegor256** [comment]
@dmzaytsev we'll use `toString`

**Comment by davvd** [comment]
> @davvd assign someone else pls

@yegor256 -30 points to your rating

**Comment by davvd** [comment]
> @davvd assign someone else pls

@yegor256 OK, I will try to assign someone else

**Comment by davvd** [comment]
@bdragan please go ahead, it's your task now, keep [this](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind, and don't hesitate to ask any technical questions you may have; Budget here is **30 mins** (keep [this principle](http://www.xdsd.org/2014/04/17/how-xdsd-is-different.html) in mind)

**Comment by bdragan** [comment]
@dmzaytsev, @yegor256, @davvd  I would suggest using Lombok `@ToString` for this purpose.

Then, for a `Take` like:
    new TkWithType(new TkClasspath(), "text/xsl")
the output would be:
    TkWithType(super=TkWrap(origin=org.takes.tk.TkWithType$1@6b601188))
and for a `Take` like:
    new TkFlash(new TkForward(new TkFork(new FkRegex("/", new TkPostMessage()))))
the output is:
    TkFlash(origin=TkForward(origin=TkFork(forks=[org.takes.facets.fork.FkRegex@7a2ad171])), cookie=RsFlash)

As you see, classes that don't provide the `toString` method (including the `Take`s implemented as anonymous classes) are still printed using the default `toString` method. Nevertheless, I think we should not exclude these from the toString output (at least the type name is printed, and we can later introduce the proper `toString` implementations for them as well if needed).

Would you agree?

Also, circular references can be problematic. At the first glance, I would say that there should not be any when it comes to `Take`s and `Response`s, am I right?

**Comment by yegor256** [comment]
@bdragan looks good to me

**Comment by bdragan** [comment]
@davvd Pull Request #398 created and ready for review.

**Comment by davvd** [comment]
> @davvd Pull Request #398 created and ready for review.

@bdragan thanks, we'll review it soon

**Comment by bdragan** [comment]
@davvd This resolution of this issue is currently blocked by #403.

**Comment by davvd** [comment]
> @davvd This resolution of this issue is currently blocked by #403.

@bdragan ok, let's wait for #403

**Comment by davvd** [comment]
@dmzaytsev we waited for #403, it is closed already

**Comment by bdragan** [comment]
@dmzaytsev @davvd PR #398 has been merged to master.

**Comment by dmzaytsev** [comment]
@bdragan thanks!

**Comment by davvd** [comment]
@ypshenychka please, let us know what do you think about this ticket, according to our [QA rules](http://www.teamed.io/qa.html)

**Comment by ypshenychka** [comment]
@davvd 
>davvd commented on 3 Apr
the code made here contains a puzzle 101-0be068b8, which will be resolved soon

According to our [QA Rules](http://www.teamed.io/qa.html):
>Messages in a ticket always start with a name of a user they are addressed to.

Please correct your message by indicating an addressee in the beginning.

**Comment by ypshenychka** [comment]
@davvd Please remove assignment from @yegor256 as far as we have label with @bdragan who actually fixed the issue.

**Comment by davvd** [comment]
> @davvd 
> >davvd commented on 3 Apr
> the code made here contains a puzzle 101-0be068b8, which will be resolved soon
> 
> According to our [QA Rules](http://www.teamed.io/qa.html):
> >Messages in a ticket always start with a name of a user they are addressed to.
> 
> Please correct your message by indicating an addressee in the beginning.

@ypshenychka @yegor256 what do you think?

**Comment by yegor256** [comment]
@davvd asskgin @bdragan here pls

**Comment by davvd** [comment]
> @davvd assign @bdragan here pls

@yegor256 done, @bdragan is assigned to this task, and is already working on it, AFAIK

**Comment by ypshenychka** [comment]
@davvd Please remove assignment from @yegor256 and also correct your message (as mentioned in above [remark](https://github.com/yegor256/takes/issues/101#issuecomment-148908892)).

**Comment by davvd** [comment]
> @davvd Please remove assignment from @yegor256 and also correct your message (as mentioned in above [remark](https://github.com/yegor256/takes/issues/101#issuecomment-148908892)).

@ypshenychka don't pay attention to the assignee, it's a technical issue on our side

**Comment by ypshenychka** [comment]
@davvd Please correct your message by indicating an addressee in the beginning.
>davvd commented on 3 Apr
the code made here contains a puzzle 101-0be068b8, which will be resolved soon

**Comment by davvd** [comment]
> @davvd Please correct your message by indicating an addressee in the beginning.
> >davvd commented on 3 Apr
> the code made here contains a puzzle 101-0be068b8, which will be resolved soon

@ypshenychka fixed, thanks for noticing

**Comment by ypshenychka** [comment]
@davvd Thank you.

**Comment by ypshenychka** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Thank you.

@ypshenychka no problem :)

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@ypshenychka thanks for the QA review, we'll work better next time

**Comment by davvd** [comment]
@bdragan paied 10 mins to @ypshenychka for QA review (payment ID is `68465482`). Thank you, I have added **1 hour** to your account in payment/transaction "`68465485`", time consumed: zero. there is a bonus for fast delivery (m=0). added +60 to your rating, now it is equal to [+143](http://www.netbout.com/b/35709?open=rating)

**Comment by davvd** [comment]
@dmzaytsev last puzzled solved `101-0be068b8`/#174
